src/il/tos.nim

```diff
@@ -178,35 +181,37 @@ proc `$`*(self: Pattern): string =
     of PatternKind.UnderScore:
         "_"
 proc `$`*(self: IdentDef): string =
-    case self.kind
-    of DefKind.Def:
-        let
-            pat = $self.pat
-            typ = if self.typ.isNone: "" else: fmt": {self.typ.get}"
-            default = if self.default.isNone: "" else: fmt" = {self.default.get}"
-            docStr = if self.docStr.isNone: "" else: fmt" #{self.docStr.get}"
-        pat & typ & default & docStr
-    of DefKind.Comment:
-        "#" & self.comment
+    let
+        pat = $self.pat
+        typ = if self.typ.isNone: "" else: fmt": {self.typ.get}"
+        default = if self.default.isNone: "" else: fmt" = {self.default.get}"
+        docStrs = self.comments.filterIt(it.isDoc)
+        comments = self.comments.filterIt(not it.isDoc)
+    assert docStrs.len <= 1
```

Comment:
what should docStr's length be less than 1 for?

Reply:
i'v thought that only the comment that is in the same line as the pattern is docStr and the others are (ordinary) comments.
so, i assumed the number of docStr would be 0 or 1.